### PR TITLE
Keycloak integration

### PR DESCRIPTION
**Pull request by xcrou89**
@rasheedamir This is the initial commit for keycloak config and customer implementation. Please check. Thanks

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+KEYCLOAK_REALM=nordmart
+KEYCLOAK_URL=http://192.168.99.100:8180/auth/
+KEYCLOAK_CLIENT_BACKEND=stakater-nordmart-backend
+KEYCLOAK_CLIENT_SECRET_KEY=7672aea2-81ff-4aa2-affe-daf3bad10334
```
Comment (ahmedwaleedmalik) [inline .env]: Use `application.properties` ? No hard-coding of URLs

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    builderImage = "stakater/builder-maven:3.5.4-jdk1.8-apline8-v0.0.3"
+    goal = "clean package"
+    notifySlack = true
+    runIntegrationTest = false
+    gitUser = "stakater-user"
+    gitEmail = "stakater@gmail.com"
+    usePersonalAccessToken = true
+    tokenCredentialID = 'GithubToken'
+    serviceAccount = "jenkins"
+    dockerRepositoryURL = 'docker-delivery.cp-stakater.com:443'
```
Comment (ahmedwaleedmalik) [inline Jenkinsfile]: please replace this with `docker.delivery.stakater.com:443`

File: deployment/values.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    tag: v0.0.1
+  env:
+  - name: KEYCLOAK_REALM
+    value: "nordmart"
+  - name: KEYCLOAK_URL
+    value: "http://192.168.99.100:8180/auth/"
+  - name: KEYCLOAK_CLIENT_BACKEND
+    value: "stakater-nordmart-backend"
+  - name: KEYCLOAK_CLIENT_SECRET_KEY
+    value: "7672aea2-81ff-4aa2-affe-daf3bad10334"
```
Comment (ahmedwaleedmalik) [inline deployment/values.yaml]: No hard-coding, same feedback as https://github.com/stakater-lab/stakater-nordmart-cart/pull/26/files

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: deployment/values.yaml (hunk 3 above)
Comment (xcrou89) [reply deployment/values.yaml]: Fixed. I added default values

File: Jenkinsfile (hunk 2 above)
Comment (xcrou89) [reply Jenkinsfile]: Done.

File: .env (hunk 1 above)
Comment (xcrou89) [reply .env]: Done.

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/customer:0.0.0-PR-1-1-SNAPSHOT`

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/customer:0.0.0-PR-1-2-SNAPSHOT`

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/customer:0.0.0-PR-1-3-SNAPSHOT`

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]
Caused by: java.lang.RuntimeException: RESTEASY003325: Failed to construct public org.keycloak.services.resources.KeycloakApplication(javax.servlet.ServletContext,org.jboss.resteasy.core.Dispatcher)
	at org.jboss.resteasy.core.ConstructorInjectorImpl.construct(ConstructorInjectorImpl.java:164)
	at org.jboss.resteasy.spi.ResteasyProviderFactory.createProviderInstance(ResteasyProviderFactory.java:2750)
	at org.jboss.resteasy.spi.ResteasyDeployment.createApplication(ResteasyDeployment.java:364)
	at org.jboss.resteasy.spi.ResteasyDeployment.startInternal(ResteasyDeployment.java:277)
	at org.jboss.resteasy.spi.ResteasyDeployment.start(ResteasyDeployment.java:89)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.init(ServletContainerDispatcher.java:119)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.init(HttpServletDispatcher.java:36)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:117)
	at org.wildfly.extension.undertow.security.RunAsLifecycleInterceptor.init(RunAsLifecycleInterceptor.java:78)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:103)
	at io.undertow.servlet.core.ManagedServlet$DefaultInstanceStrategy.start(ManagedServlet.java:303)
	at io.undertow.servlet.core.ManagedServlet.createServlet(ManagedServlet.java:143)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:583)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:554)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at io.undertow.servlet.core.DeploymentManagerImpl.start(DeploymentManagerImpl.java:596)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:97)


Getting the following error when i try to initiate keycloak with the "realm-config.json"

**Comment by xcrou89** [comment]
@ahmedwaleedmalik I could not reproduce the issue. Make sure the docker can hit the configuration file (when mount the file realm-config.json).

**Comment by stakater-user** [comment]
@xcrou89 Yikes! You better fix it before anyone else finds out! [Build 4](https://jenkins.release.stakater.com/job/nordmart-apps/job/stakater-nordmart-customer/job/PR-1/4/) has Failed!

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/customer:0.0.0-PR-1-7-SNAPSHOT`

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/customer:0.0.0-PR-1-8-SNAPSHOT`

**Review by ahmedwaleedmalik — APPROVED** [review]

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/customer:0.0.0-PR-1-9-SNAPSHOT`

**Pull request merged**